### PR TITLE
Fix volume REST test

### PR DESCRIPTION
**Pull request by kotfic**
Fix attach volume test
Fix create volume tests
Fix delete volume test
Fix detach volume test
Fix find volume test
Fix get status volume test
Fix get volume test
Remove skips, fix static analysis

### CODE REVIEW

**Comment by kotfic** [comment]
@cjh1 PTAL

File: cumulus/ansible/tasks/providers/ec2.py [hunk 1]
```diff
@@ -38,7 +38,11 @@ class EC2Provider(CloudProvider):
     }
 
     def __init__(self, profile):
-        super(EC2Provider, self).__init__(profile)
+        # super(EC2Provider, self).__init__(profile)
+        self.girder_profile_id = profile.get('_id', None)
+
+        for key, value in profile.items():
+            setattr(self, key, value)
```
Comment (cjh1) [inline cumulus/ansible/tasks/providers/ec2.py]: Why this change? Also if we don't need 41 any more lets remove it.

File: girder/cumulus/plugin_tests/volume_test.py [hunk 2]
```diff
@@ -179,27 +167,41 @@ def test_create(self, get_ec2_client):
...
             u'fs': u'ext4',
             u'ec2': {
-                u'id': volume_id
+                u'id': None
             },
             u'profileId': self._profile_id
         }
+
         del r.json['_id']
+        # import rpdb; rpdb.set_trace()
```
Comment (cjh1) [inline girder/cumulus/plugin_tests/volume_test.py]: Left over debugging

File: girder/cumulus/plugin_tests/volume_test.py [hunk 3]
```diff
@@ -302,13 +299,20 @@ def test_get(self, get_ec2_client):
...
-        volume_id = 'vol-1'
-        ec2_client = get_ec2_client.return_value
-        ec2_client.create_volume.return_value = {
-            'VolumeId': volume_id
+    @mock.patch('girder.plugins.cumulus.volume.CloudProvider')
+    @mock.patch('cumulus.ansible.tasks.volume.attach_volume.delay')
+    @mock.patch('cumulus.ansible.tasks.volume.detach_volume.delay')
+    @mock.patch('cumulus.ansible.tasks.volume.delete_volume.delay')
+    def test_delete(self, delete_volume, detach_volume,
+                    attach_volume, CloudProvider):
```
Comment (cjh1) [inline girder/cumulus/plugin_tests/volume_test.py]: Lets validate that the delete_volume function is called.

File: girder/cumulus/plugin_tests/volume_test.py [hunk 4]
```diff
@@ -323,64 +327,82 @@ def test_delete(self, get_ec2_client):
...
         self.assertStatusOk(r)
 
         r = self.request('/volumes/%s' % volume_id, method='DELETE',
                          user=self._user)
         self.assertStatus(r, 200)
 
+    @mock.patch('girder.plugins.cumulus.volume.CloudProvider')
+    @mock.patch('cumulus.ansible.tasks.volume.attach_volume.delay')
+    def test_attach_volume(self, attach_volume, CloudProvider):
+#        import rpdb; rpdb.set_trace()
```
Comment (cjh1) [inline girder/cumulus/plugin_tests/volume_test.py]: More left over debug ...

File: girder/cumulus/server/volume.py [hunk 5]
```diff
@@ -108,6 +108,10 @@ def patch(self, volume, params):
     @access.user
     def create(self, params):
         body = getBodyJson()
+
+#        if body.get("name", None) == 'test2':
+#            import rpdb; rpdb.set_trace()
+
```
Comment (cjh1) [inline girder/cumulus/server/volume.py]: Remove?

File: cumulus/ansible/tasks/providers/ec2.py (hunk 1 above)
Comment (kotfic) [inline cumulus/ansible/tasks/providers/ec2.py]: Code from the parent class was moved into the EC2Provider - calling init like this was causing all kinds of issues with mocking.

**Comment by kotfic** [comment]
@cjh1 PTAL
